dev/NavigationView/NavigationView.cpp

```diff
@@ -688,14 +700,46 @@ void NavigationView::UpdateTopNavRepeatersItemSource(const winrt::IInspectable&
     m_topDataProvider.SetDataSource(items);
 
     // rebinding
+    UpdateTopNavRepeaterItemsSource(items);
+    UpdateTopNavOverflowRepeaterItemsSource(items);
+}
+
+void NavigationView::UpdateTopNavRepeaterItemsSource(const winrt::IInspectable& items)
+{
     if (items)
     {
         UpdateItemsRepeaterItemsSource(m_topNavRepeater.get(), m_topDataProvider.GetPrimaryItems());
-        UpdateItemsRepeaterItemsSource(m_topNavRepeaterOverflowView.get(), m_topDataProvider.GetOverflowItems());
     }
     else
     {
         UpdateItemsRepeaterItemsSource(m_topNavRepeater.get(), nullptr);
+    }
+}
+
+void NavigationView::UpdateTopNavOverflowRepeaterItemsSource(const winrt::IInspectable& items)
+{
+    m_topNavOverflowItemsCollectionChangedRevoker.revoke();
+
+    if (items)
+    {
+        UpdateItemsRepeaterItemsSource(m_topNavRepeaterOverflowView.get(), m_topDataProvider.GetOverflowItems());
```

Comment:
>UpdateItemsRepeaterItemsSource(m_topNavRepeaterOverflowView.get(), m_topDataProvider.GetOverflowItems()); [](start = 8, length = 105)

I'd suggest moving this into the if block below, avoids calling the trackerref get twice.

Reply:
@StephenLPeters In that case, we could also decide to "inline" UpdateItemsRepeaterItemsSource() because that also checks if the passed repeater (overflowRepeater = m_topNavRepeaterOverflowView.get() here) is not `null` so we could skip yet another duplicated check. 

That said, I initially used an "inlined" version but I felt it didn't really resemble the layout of the related `UpdateTopNavRepeaterItemsSource()` implementation:

```cpp
void NavigationView::UpdateTopNavOverflowRepeaterItemsSource(const winrt::IInspectable& items)
{
    m_topNavOverflowItemsCollectionChangedRevoker.revoke();

    if (const auto overflowRepeater = m_topNavRepeaterOverflowView.get())
    {
        if (items)
        {
            const auto itemsSource = m_topDataProvider.GetOverflowItems();
            overflowRepeater.ItemsSource(itemsSource);

            m_topNavOverflowItemsCollectionChangedRevoker = overflowRepeater.ItemsSourceView().CollectionChanged(winrt::auto_revoke, { this, &NavigationView::OnOverflowItemsSourceCollectionChanged });
        }
        else
        {
            overflowRepeater.ItemsSource(nullptr);
        }
    }
}
```
As you can see, I "inlined" UpdateItemsRepeaterItemsSource() above thus skipping the null check it does on the passed ItemsRepeater (as we already do that here).

For similarity purposes with UpdateTopNavOverflowRepeaterItemsSource() (the function you are suggestiong a name change for) I decided to to not inline UpdateItemsRepeaterItemsSource() here even though that means we are doing duplicated null checks. Those normally shouldn't be run in a tight loop though so it appears fine to me here.

Your suggestion here could be a good middleground between both similarity with related implementations and still looking after great performance. Though I can also use the inlined version above if the WinUI team prefers that.

Reply:
@StephenLPeters Once you gave your thoughts on this I will create a PR addressing your feedback.

Reply:
I think i have a slight preference to your inlined version, but they are similar.

Reply:
Added the inlined version.